doc/adrs/069_add_support_for_children_layout_strategy.adoc

```diff
@@ -0,0 +1,103 @@
+= ADR-069 - Add support for children layout strategy
+
+== Context
+
+We want to improve layout engines (ELK and Incremental) to make them more powerful. We need them to be more powerful to support existing node types and two new types of node, compartment and svg image.
+
+== Decision
+
+We have chosen to separate the behavior that does the layout of a node from the behavior that does the layout of the node children.
+
+The existing node style which is used to handle list node, list item node, rectangular node and image node will be split into three types of nodes:
+- rectangular node style (has an option to consider the label is inside a header),
+- image node style (has an attribute to define the image),
+- icon label node style.
+Each node style will be used to describe how the node is represented in the diagram.
+
+We will introduce the children layout strategy to describe how the layout of node children is done.
+At first, we will support two kind of children layout strategy:
+- free form layout strategy
+- list layout strategy
+- no child layout strategy
+
+For the _list layout strategy_, it will be possible to describe:
```

Comment:
Splitting the two aspects opens the possibility for a lot a combinations (that's the point), but this is a single example.

What about the other combinations? In particular, are there combinations that are not supported/do not make sense, and what's the behavior in this case. For example I guess with "icon label node style" the only strategy that makes sense is "no child", but it should be said explicitly.

What happens if I have:
* parent node A (rectangular, free form)
* a child B of style "icon label"
i.e. how is a, "icon label" displayed inside a free form container?

What if I have:
* parent node A (rectangular, list)
* child node B (image, free form)
i.e. how is an "image" node displayed when inside a list?

etc.

Reply:
It seems this would allow for a node with an image background to have children nodes (as list items or sub-nodes in free form). Is this explicitly supported?

Reply:
I added a table with all the possibilities in the ADR.
Node with an image background will support children nodes if the image is a SVG.